app/models/server/models/Rooms.js

```diff
@@ -552,38 +552,40 @@ export class Rooms extends Base {
 		return this._db.find(query, options);
 	}
 
-	findByNameOrFNameAndTypeIncludingTeamRooms(name, type, teamIds, options) {
+	findByNameOrFNameAndRoomIdsIncludingTeamRooms(text, teamIds, roomIds, options) {
+		const searchTerm = text && new RegExp(text, 'i');
+
 		const query = {
-			t: type,
-			teamMain: {
-				$exists: false,
-			},
 			$and: [
+				{ teamMain: { $exists: false } },
+				{ prid: { $exists: false } },
 				{
 					$or: [
 						{
-							teamId: {
-								$exists: false,
-							},
+							t: 'c',
+							teamId: { $exists: false },
 						},
 						{
-							teamId: {
-								$in: teamIds,
-							},
+							t: 'c',
+							teamId: { $in: teamIds },
 						},
+						...roomIds?.length > 0 ? [{
+							_id: {
+								$in: roomIds,
+							},
+						}] : [],
 					],
 				},
-				{
+				...searchTerm ? [{
```

Comment:
Do you need this check? Shouldn't an empty regex match everything anyways? 👀

Reply:
indeed it matches everything anyways, but I wonder if the database will actually try to match everything against the RegExp, so since we know we want everything I _think_ it is better to not even send it to db

Reply:
🤔 that's a fair point. It looks like for a small collection with index, it'll prefer the IXSCAN, while for bigger ones, it will only attempt to use the index when the regex is /^.../

Agree that it's better to not to send it at all.